delivery/common/jwt/lib/jwt.js

```diff
@@ -0,0 +1,94 @@
+        if (this.jwtOptions.subject && jwtPayload.sub && this.jwtOptions.subject !== jwtPayload.sub) throw new Error(`JWT malformed: invalid sub, expected ${this.jwtOptions.subject}`);
+        if (this.jwtOptions.audience) {
+            const aud = jwtPayload.aud;
+            if (aud && (Array.isArray(aud) || "string" == typeof aud)) {
+                if (!(Array.isArray(aud) ? aud : [ aud ]).includes(this.jwtOptions.audience)) throw new Error(`JWT malformed: invalid aud, expected ${this.jwtOptions.audience}`);
+            }
+        }
+        const clockTimestamp = Math.floor(Date.now() / 1e3);
+        if (!1 === this.jwtOptions.ignoreExpiration) {
+            if ("number" != typeof jwtPayload.exp) throw new Error("JWT malformed: exp must be number");
+            if (jwtPayload.exp && clockTimestamp > jwtPayload.exp + (this.jwtOptions.clockTolerance || 0)) throw new Error("JWT expired");
+        }
+        if (!1 === this.jwtOptions.ignoreNotBefore) {
+            if ("number" != typeof jwtPayload.nbf) throw new Error("JWT malformed: nbf must be number");
+            if (jwtPayload.nbf && jwtPayload.nbf > clockTimestamp + (this.jwtOptions.clockTolerance || 0)) throw new Error("JWT not active");
+        }
+        return {
+            header: jwtHeader,
+            payload: jwtPayload
+        };
+    }
+    validateOptionTypes() {
+        if (void 0 !== this.jwtOptions.issuer && ("string" != typeof this.jwtOptions.issuer || 0 === this.jwtOptions.issuer.trim().length)) throw new Error("Invalid jwtOptions: issuer must be non empty string");
+        if (void 0 !== this.jwtOptions.audience && ("string" != typeof this.jwtOptions.audience || 0 === this.jwtOptions.audience.trim().length)) throw new Error("Invalid jwtOptions: audience must be non empty string");
+        if (void 0 !== this.jwtOptions.subject && ("string" != typeof this.jwtOptions.subject || 0 === this.jwtOptions.subject.trim().length)) throw new Error("Invalid jwtOptions: subject must be non empty string");
+        if (void 0 === this.jwtOptions.ignoreExpiration) this.jwtOptions.ignoreExpiration = !0; else if ("boolean" != typeof this.jwtOptions.ignoreExpiration) throw new Error("Invalid jwtOptions: ignoreExpiration must be boolean");
+        if (void 0 === this.jwtOptions.ignoreNotBefore) this.jwtOptions.ignoreNotBefore = !0; else if ("boolean" != typeof this.jwtOptions.ignoreNotBefore) throw new Error("Invalid jwtOptions: ignoreNotBefore must be boolean");
+        if (void 0 === this.jwtOptions.clockTolerance) this.jwtOptions.clockTolerance = 60; else if ("number" != typeof this.jwtOptions.clockTolerance) throw new Error("Invalid jwtOptions: clockTimestamp must be number");
+    }
+    async validate(base64JWTToken, alg, key) {
+        if ("string" != typeof base64JWTToken || "string" != typeof alg || "string" != typeof key) throw new Error("Invalid arguments!");
+        const jwtParts = base64JWTToken.split(".");
+        switch (alg) {
+          case "RS256":
```

Comment:
Is there a reason not to support the other key lengths? 384 and 512 bit keys would be handy.

Reply:
@evan-hughes As of now, the jwt module was implemented as per the requirement for watermarking use cases. The use case only needed support for HS256 and RS256.

Reply:
All key lengths are supported.